libs/accounts/recovery-phone/src/lib/sms.manager.ts

```diff
@@ -73,12 +87,35 @@ export class SmsManager {
       });
       return msg;
     } catch (err) {
+      if (err.code === TwilioErrorCodes.SMS_SEND_RATE_LIMIT_EXCEEDED) {
+        if (retryCount < this.config.maxRetries) {
+          // Exp back off and try again
+          await new Promise((r) =>
+            setTimeout(r, Math.pow(2, retryCount++) * 1000)
+          );
+          return await this._sendSMS(to, body, uid, retryCount);
+        }
+      }
+
       this.metrics.increment('sms.send.error');
 
       if (err.code === TwilioErrorCodes.INVALID_TO_PHONE_NUMBER) {
         throw new RecoveryNumberInvalidFormatError(uid || '', to, err);
       }
+      if (err.code === TwilioErrorCodes.SMS_SEND_RATE_LIMIT_EXCEEDED) {
+        throw new SmsSendRateLimitExceededError(uid || '', to, from, err);
+      }
       throw err;
     }
   }
+
+  public rotateFromNumber() {
+    if (typeof this.config.from === 'string') {
+      return this.config.from;
```

Comment:
Is this ever going to be a string with your update that sets it to `Array`? Just curious, fine by me to keep it as a fallback.

Reply:
Ah, good catch. I was thinking it could be either, but for the convict config it's better to just have a single type. I'll fix this.